src/trees/tree.ts

```diff
@@ -0,0 +1,84 @@
+import {Component, ContentChild, ContentChildren, QueryList, AfterContentInit, OnDestroy, HostListener} from 'angular2/core';
+import {NglTreeRegistrationService} from './tree-registration.service';
+import {NglTreeHeading} from './tree-heading';
+import {NglTreeBranch} from './tree-branch';
+import {NglSubTree} from './sub-tree';
+import {uniqueId} from '../util/util';
+
+@Component({
+  selector: 'ngl-tree',
+  templateUrl: './tree.jade',
+  host: {
+    'tabindex': '-1',
+  },
+  providers: [NglTreeRegistrationService],
+})
+export class NglTree implements AfterContentInit, OnDestroy {
+  @ContentChild(NglTreeHeading) myHeading: NglTreeHeading;
+  @ContentChildren(NglTreeBranch) myBranches = new QueryList<NglTreeBranch>();
+  @ContentChildren(NglSubTree) mySubTrees = new QueryList<NglSubTree>();
+
+  private children: any[] = [];
+  private level = 1;
+  /* tslint:disable */
+  private uniqueId = uniqueId();
```

Comment:
Why is the linter disabled here? Shouldn't adding uniqueId be moved to constructor?